netmap/types.proto

```diff
@@ -5,89 +5,96 @@ package neo.fs.v2.netmap;
 option go_package = "github.com/nspcc-dev/neofs-api-go/v2/netmap/grpc;netmap";
 option csharp_namespace = "NeoFS.API.v2.Netmap";
 
+// Operations on filters
+enum Operation {
+  // No Operation defined
+  OPERATION_UNSPECIFIED= 0;
+
+  // Equal
+  EQ = 1;
+
+  // Not Equal
+  NE = 2;
+
+  // Greater then
+  GT = 3;
+
+  // Greater or equal
+  GE = 4;
+
+  // Less then
+  LT = 5;
+
+  // Less or equal
+  LE = 6;
+
+  // Logical OR
+  OR = 7;
+
+  // Logical AND
+  AND = 8;
+}
+
+// Filter
+message Filter {
+  // Name of the filter or a reference to the named filter.
+  // '*' means application to the whole unfiltered NetworkMap
+  string name = 1;
+
+  // Key to filter
+  string key = 2;
+
+  // Filtering operation
+  Operation op = 3;
+
+  // Value to match
+  string value = 4;
+
+  // Next Filter logical relation
+  Operation next_op = 5;
```

Comment:
Shouldn't this be also a `Filter`? What value does this operation applies to?

Reply:
That's exactly not to use Filter here and have chain of Filters without recursion.

Reply:
Does the semantics of `(a AND) (b OR) (c AND) (d *)` correspond to the DNF `(a AND b) OR (c AND d)` or to the `((a AND b) OR c) AND d`?